### ISSUE TITLE
Econml summary() result interpretation

### ISSUE THREAD

**Issue by itewqq** [microsoft/EconML#650]
I just started using EconML library, I would like to ask what does the output of inference summary mean? what exactly is the prediction/inference doing here? Have searched through the doc, but find no answer. 

Also, I don't understand about the p-value, isn't it a prediction here? Why use the metric from the hypothesis test?

```
   point_estimate  stderr  zstat  pvalue  ci_lower  ci_upper
X                                                           
0          -0.602   0.106 -5.672   0.000    -0.776    -0.427
1          -0.042   0.085 -0.487   0.627    -0.182     0.099
2          -0.161   0.125 -1.295   0.195    -0.366     0.044
```

Why would there be a statistic like stderr for each row (`each sample`)? My understanding an single sample does not have a `stderr`, so with whom is it compared and calculated?

My test code, mostly copied from official examples:

```python
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split


# DoubleML 
from scipy.linalg import toeplitz

from sklearn.preprocessing import PolynomialFeatures, OneHotEncoder
from sklearn.datasets import make_spd_matrix

from doubleml import DoubleMLData


# EconML 
from econml.dml import LinearDML
from sklearn.linear_model import (Lasso, LassoCV, LogisticRegression,
                                  LogisticRegressionCV,LinearRegression,
                                  MultiTaskElasticNet,MultiTaskElasticNetCV)
from sklearn.ensemble import RandomForestRegressor,RandomForestClassifier
from sklearn.preprocessing import PolynomialFeatures
from econml.inference import BootstrapInference
from itertools import product
# DATA: Simulate data

# Treatment effect function
def exp_te(x):
    return np.exp(2*x[0])

# DGP constants
np.random.seed(123)
n = 2000
n_w = 30
support_size = 5
n_x = 10
# Outcome support
support_Y = np.random.choice(np.arange(n_w), size=support_size, replace=False)
coefs_Y = np.random.uniform(0, 1, size=support_size)
epsilon_sample = lambda n: np.random.uniform(-1, 1, size=n)
# Treatment support
support_T = support_Y
coefs_T = np.random.uniform(0, 1, size=support_size)
eta_sample = lambda n: np.random.uniform(-1, 1, size=n)

# Generate controls, covariates, treatments and outcomes
W = np.random.normal(0, 1, size=(n, n_w))
X = np.random.uniform(0, 1, size=(n, n_x))
# X = W

# Heterogeneous treatment effects
# TE = np.array([exp_te(x_i) for x_i in X])
TE = 0.2 + 0.3 * X[:,0] - 0.2 * X[:,2] + X[:,3] - 2 * X[:,5] + 0.2 * X[:,9]
T = np.dot(W[:, support_T], coefs_T) + eta_sample(n)
Y = TE * T + np.dot(W[:, support_Y], coefs_Y) + epsilon_sample(n)

Y_train, Y_val, T_train, T_val, X_train, X_val, W_train, W_val = train_test_split(Y, T, X, W, test_size=.2)
# Generate test data
# X_test = np.array(list(product(np.arange(0, 1, 0.01), repeat=n_x)))
X_test = np.random.uniform(0, 1, size=(3, n_x))
# Estimation Methods: simulation
# est = LinearDML(model_y=LassoCV(),model_t=LassoCV())
est = LinearDML(model_y=RandomForestRegressor(),
                model_t=RandomForestRegressor(),
                random_state=123)

### Estimate with OLS confidence intervals

#simulation
est.fit(Y_train, T_train, X=X_train, W=W_train)
te_pred = est.effect(X_test)
lb, ub = est.effect_interval(X_test, alpha=0.05) # OLS confidence intervals

pnt_coef = est.coef_
lb_coef,ub_coef = est.coef__interval(alpha=.05)

# print(f"{te_pred},\n,{lb},\n,{ub}")
# print()
# print()
# # print(f"{pnt_coef},\n,{lb_coef},\n,{ub_coef}")

print(pnt_coef)

print(est.summary())

# x_0 0.3 , x_2 -0.2, x_3 1,  x_5 -2, x_9 0.2
print(est.effect_inference(X_test).summary_frame(alpha=0.1,value=0,decimals=3))
```

**Comment by kbattocchi** [comment]
Perhaps this analogy will be helpful: if you want to estimate a linear equation (Y=X*β+error), you can take a bunch of samples of features and corresponding outputs and perform a linear regression, which can give you both a point estimate of the coefficient β as well as confidence intervals based on the variation in the inputs.  Then, if you want to predict what value Y would take given some value of X, you can also get a confidence interval for the prediction, based on the confidence intervals for the coefficients.

That's basically the same idea as what we're doing here: we're trying to estimate a more complicated equation (trying to determine θ(X) based on the equation Y=T*θ(X)+f(X,W) plus another equation for T), but based on certain assumptions we can calculate not only a point value for the coefficients determining θ but also confidence intervals, and then we can extend that to confidence intervals for the prediction θ(X) for each input X.  The p-value then corresponds to a hypothesis test comparing the quantity to zero (e.g. a very low p-value is inconsistent with the quantity that we're trying to calculate being exactly zero, given the assumptions of the model).

Does that help?

**Comment by itewqq** [comment]
Thank you very much for the explanation, It was super helpful!

However
> (e.g. a very low p-value is inconsistent with the quantity that we're trying to calculate being exactly zero, given the assumptions of the model).

May I ask more about where the `exactly zero` comes from here?

**Comment by kbattocchi** [comment]
I just mean that the hypothesis being tested is that the quantity is 0; however, if the coefficient is only slightly different from zero at the scale that is important for your application, then as the number of samples increases we will still _eventually_ get small p-values because the confidence intervals should keep shrinking around the ever-more-precisely estimated non-zero value.

In other words, the p-value measures the statistical significance of the difference from zero, but not necessarily the significance in real terms for whatever application you're using it on.

**Comment by GladysKao** [comment]
> Perhaps this analogy will be helpful: if you want to estimate a linear equation (Y=X*β+error), you can take a bunch of samples of features and corresponding outputs and perform a linear regression, which can give you both a point estimate of the coefficient β as well as confidence intervals based on the variation in the inputs. Then, if you want to predict what value Y would take given some value of X, you can also get a confidence interval for the prediction, based on the confidence intervals for the coefficients.
> 
> That's basically the same idea as what we're doing here: we're trying to estimate a more complicated equation (trying to determine θ(X) based on the equation Y=T*θ(X)+f(X,W) plus another equation for T), but based on certain assumptions we can calculate not only a point value for the coefficients determining θ but also confidence intervals, and then we can extend that to confidence intervals for the prediction θ(X) for each input X. The p-value then corresponds to a hypothesis test comparing the quantity to zero (e.g. a very low p-value is inconsistent with the quantity that we're trying to calculate being exactly zero, given the assumptions of the model).
> Does that help?

--------------------------------------------------------
with this， can I take coef_ as CATE?
